### PR TITLE
Add runs endpoint

### PR DESCRIPTION
**Pull request by wgresshoff**
:heart: Thank you for your contribution!

### Description

> Add runs endpoints.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_jobs/resources/config.py [hunk 1]
```diff
@@ -37,15 +38,40 @@ class JobsResourceConfig(ResourceConfig, ConfiguratorMixin):
...
+    routes = {
+        "runs": "/jobs/<job_id>/runs",
+        "run_item": "/jobs/<job_id>/runs/<run_id>",
+        "logs": "/jobs/<job_id>/runs/<run_id>/logs",
+    }
+
+    # Request parsing
+    request_view_args = {
+        "job_id": ma.fields.Int(),
+        "run_id": ma.fields.Int(),
```
Comment (slint) [inline invenio_jobs/resources/config.py:71]:
```suggestion
        "job_id": ma.fields.UUID(),
        "run_id": ma.fields.UUID(),
```

minor: to match the datamodel

File: invenio_jobs/resources/config.py [hunk 2]
```diff
@@ -37,15 +38,40 @@ class JobsResourceConfig(ResourceConfig, ConfiguratorMixin):
...
+    """Runs resource config."""
+
+    # Blueprint configuration
+    blueprint_name = "job_runs"
+    url_prefix = ""
+
+    routes = {
+        "runs": "/jobs/<job_id>/runs",
+        "run_item": "/jobs/<job_id>/runs/<run_id>",
+        "logs": "/jobs/<job_id>/runs/<run_id>/logs",
```
Comment (slint) [inline invenio_jobs/resources/config.py:65]:
```suggestion
        "list": "/jobs/<job_id>/runs",
        "item": "/jobs/<job_id>/runs/<run_id>",
        "logs_list": "/jobs/<job_id>/runs/<run_id>/logs",
```

nit/minor: to follow convention

File: invenio_jobs/resources/config.py [hunk 3]
```diff
@@ -37,15 +38,40 @@ class JobsResourceConfig(ResourceConfig, ConfiguratorMixin):
...
+    }
```
Comment (slint) [inline invenio_jobs/resources/config.py:66]:
```suggestion
        "actions_stop": "/jobs/<job_id>/runs/<run_id>/actions/stop",
    }
```

minor: to have an explicit endpoint for stopping/cancelling a task

File: invenio_jobs/services/config.py [hunk 4]
```diff
@@ -73,3 +74,36 @@ class JobsServiceConfig(ServiceConfig, ConfiguratorMixin):
...
+    search = JobSearchOptions
+
+    # Service schema
+    schema = JobSchema
+
+    links_item = {
+        "self": JobLink("{+api}/jobs/{id}/runs"),
+        "start": JobLink("{+api}/jobs/{id}/runs"),
+        "delete": JobLink("{+api}/jobs/{id}/runs/{run_id}"),
+        "logs": JobLink("{+api}/jobs/{id}/runs/{run_id}/logs"),
```
Comment (slint) [inline invenio_jobs/services/config.py:108]:
```suggestion
        "self": JobLink("{+api}/jobs/{job_id}/runs/{run_id}"),
        "stop": JobLink("{+api}/jobs/{job_id}/runs/{run_id}/actions/stop"),
        "logs": JobLink("{+api}/jobs/{job_id}/runs/{run_id}/logs"),
```

- Renamed `id` to `job_id`
- Added `stop` link

**Review by slint — APPROVED** [review]
LGTM, some minor comments. If possible also run `black` and `isort` to fix the linter errors.

File: invenio_jobs/resources/resources.py
Comment (carlinmack) [inline invenio_jobs/resources/resources.py:1]: maybe you want to add Münster to the copyright statement

**Review by carlinmack — APPROVED** [review]

File: invenio_jobs/resources/resources.py
Comment (carlinmack) [reply invenio_jobs/resources/resources.py:1]: and running `black .` and `isort .` would be great :)

File: invenio_jobs/resources/config.py (hunk 1 above)
Comment (wgresshoff) [reply invenio_jobs/resources/config.py]: Fixed.

File: invenio_jobs/resources/config.py (hunk 2 above)
Comment (wgresshoff) [reply invenio_jobs/resources/config.py]: Fixed.

File: invenio_jobs/services/config.py (hunk 4 above)
Comment (wgresshoff) [reply invenio_jobs/services/config.py]: Fixed.

File: invenio_jobs/resources/config.py (hunk 3 above)
Comment (Samk13) [reply invenio_jobs/resources/config.py:67]: Should we consider adding a `pause` endpoint for pausing a task at this point @slint ?

Comment (slint) [reply invenio_jobs/resources/config.py:67]: @Samk13 I've also logged some thoughts on how/where we could be tracking the task status [in the RFC](https://codimd.web.cern.ch/qjdmyWRvRiqujzp8BC6PgA?both#Detailed-job-status-tracking).

Comment (Samk13) [reply invenio_jobs/resources/config.py:67]: @slint  Good, if the status comes with the run payload, then it's enough for us.
Regarding the link, I can't log in with KTH institution credentials, so I can't see what's in the RFC.

Comment (slint) [reply invenio_jobs/resources/config.py:67]: @karkraeg @mkloeppe, how did you manage to login through the CERN SSO? Did you use a Google account, or eduGAIN?

Comment (karkraeg) [reply invenio_jobs/resources/config.py:67]: @slint eduGAIN works

Comment (Samk13) [reply invenio_jobs/resources/config.py:67]: @slint It is possible that KTH is not integrated with EduGain, which might explain the login issue.

**Pull request merged**